managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -729,6 +733,8 @@ private synchronized void internalAsyncAddEntry(OpAddEntry addOperation) {
         if (!beforeAddEntry(addOperation)) {
             return;
         }
+        // check if ledger can be closed due to inactivity
+        checkInactiveLedgerAndRollOver(lastAddEntryTimeMs);
```

Comment:
The implementation relies on messages to trigger ledger closure due to inactivity. While it might be a slightly heavier solution, would it make more sense to have a single scheduled task that runs in the broker service to check for inactive ledgers? Otherwise, a ledger that stops receiving messages will not close until the `managedLedgerMaxLedgerRolloverTimeMinutes` has passed.

Reply:
hmm.. actually lazy check should be enough because eventually ledger will be closed if topic in unloaded, broker restarts or topic receives a new write and that will protect topic from the issue. However, adding scheduled task will not have much harm except it might repetitive close empty ledger but might give certainty to ledger age.
I have made the change and moved check to scheduled task.